src/Our.Umbraco.AuthU/Data/Plans/RegisterOAuthTables.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Our.Umbraco.AuthU.Data.Helpers;
+using Our.Umbraco.AuthU.Data.Migrations.UmbracoDbOAuthClientStore;
+using Our.Umbraco.AuthU.Data.Migrations.UmbracoDbOAuthRefreshTokenStore;
+using Our.Umbraco.AuthU.Models;
+using System;
+using System.Linq;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Migrations;
+
+namespace Our.Umbraco.AuthU.Data.Migrations.Plans
+{
+    /// <summary>
+    /// This is the plan that runs the AuthU Migrations
+    /// </summary>
+    internal class RegisterOAuthTables : MigrationPlan
+    {
+        /// <summary>
+        /// Add the demo id if preferred
+        /// </summary>
+        /// <param name="CreateDemoClient"></param>
+        public RegisterOAuthTables(bool CreateDemoClient = false) : base("AuthU")
+        {
+
+            if (!CreateDemoClient)
+            {
+                From(InitialState)
+                .To<CreateAuthClientStoreTable>("CreateAuthClientStoreTable")
+                .To<CreateAuthRefreshTokenStoreTable>("CreateAuthRefreshTokenStoreTable")
+                .To<AddDeviceIdColumn>("AddDeviceIdColumn");
+            }
+            else
+            {
+                From(InitialState)
+                .To<CreateAuthClientStoreTable>("CreateAuthClientStoreTable")
+                .To<CreateAuthRefreshTokenStoreTable>("CreateAuthRefreshTokenStoreTable")
+                .To<AddDeviceIdColumn>("AddDeviceIdColumn")
+                .To<CreateDemoClient>("CreateDemoClient");
+            }
+        }
+
+        /// <summary>
+        /// Using the helper methid to get the AuthU Key 'Umbraco.Core.Upgrader.State+AuthU' Value
+        /// </summary>
+        public override string InitialState => AuthUKeyValueHelper.GetValue();
```

Comment:
I don't think this is quite right. I don't think you want to be looking up the current state from the DB, rather I think this is more for when you are upgrading from like v7. Given our install scripts are defensive, only applying when needed, I think we might be able to just have this as an empty string.